lit/docs/operation/tracing.lit

```diff
@@ -8,21 +8,104 @@ Tracing in Concourse enables the delivery of traces related to the internal
 processes that go into running builds, and other internal operations, breaking
 them down by time, and component.
 
-It currently only integrates with \link{Jaeger}{https://www.jaegertracing.io/},
+It currently integrates with \link{Jaeger}{https://www.jaegertracing.io/} and
+\link{Google Cloud Trace}{https://cloud.google.com/trace} (Stackdriver),
 although support for other systems is planned to expand as the underlying SDK
 (\link{OpenTelemetry}{https://opentelemetry.io/}) evolves.
 
 
 \section{
   \title{Configuring Tracing}
 
-  There's only one variable that is required to be set in order to leverage
-  Jaeger's integration with Concourse:
+  To export spans to Jaeger, specify the Thrift HTTP endpoint of the Jaeger
+  collector:
 
   \codeblock{bash}{{{
   CONCOURSE_TRACING_JAEGER_ENDPOINT=http://jaeger:14268/api/traces
   }}}
 
-  This tells Concourse how to target Jaeger's Thrift HTTP endpoint to send the
-  traces to.
+  To export spans to Google Cloud Trace, specify the GCP Project ID:
+
+  \codeblock{bash}{{{
+  CONCOURSE_TRACING_STACKDRIVER_PROJECTID=your-gcp-project-id
+  }}}
+
+  Note that suitable GCP credentials must be available, via the usual
+  \link{\code{GOOGLE_APPLICATION_CREDENTIALS} environment variable}{https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable},
+  the default location that the \code{gcloud} CLI expects, or from GCP's
+  metadata server (if Concourse is deployed on GCP).
+}
+
+\section{
+  \title{What's emitted?}
+
+  Below is a summary of the various operations that Concourse currently traces.
+  They are arranged like a call tree, so that for operation described, its
+  sub-operations are described indented immediately below.
+
+  \list{
+    \code{scanner.Run} -- One of the \reference{resource-checker-components}
+    responsible for determining which checks need to be run.
+    \list{
+      \code{scanner.check} -- This operation simply represents inserting the
+      check in the database.
```

Comment:
Not related to the docs, but out of curiosity - do we need to emit this operation if we emit `checker.Run` (i.e. could `checker.Run` follow from `scanner.Run`? what information would we lose if we didn't emit it?)

Reply:
This is a great question. To me, this is an important distinction - so much so that maybe the operation names should be changed to make it clear.

The scanner and checker are two communicating sequential processes and the `checks` table is the message queue between them. `scanner.run` is the 'send' whereas `checker.Run` is the 'receive', and these events can be separated in time - the degree to which they are separated seems to represent a kind of 'latency' golden signal of SRE, but I haven't found a good way to formalize this.

What do you think of this? Is it worth renaming `scanner.check` to something more like `scanner.createCheck` or `scanner.insertCheck`?

Reply:
I think a name like `scanner.createCheck` or `scanner.insertCheck` would be a bit more intuitive, but the docs do explain what it represents so it may not be necessary. I also like the CSP framing, I hadn't thought of Concourse like that but it makes sense.